DQMServices/FileIO/plugins/DQMFileSaverPB.cc

```diff
@@ -236,8 +236,26 @@ void DQMFileSaverPB::savePB(DQMStore* store, std::string const& filename, int ru
     histo.set_size(buffer.Length());
     histo.set_streamed_histo((void const*)buffer.Buffer(), buffer.Length());
 
-    // Save quality reports if this is not in reference section.
-    // XXX not supported by protobuf files.
+    // Save quality reports
+    for (QReport* qr : me->getQReports()) {
```

Comment:
`for (const auto& qr : me->getQReports()) { `
if you are just reading `qr`

Reply:
As promised, implemented this here: https://github.com/cms-sw/cmssw/pull/30592/files#diff-7b79ea209bca0428f38b6b7786f2220dR266